### PR TITLE
Andrewmwells/remove deprecated expr escape

### PR DESCRIPTION
**Pull request by andrewmwells-amazon**
## Description of changes

Update integration tests to remove `__expr` escapes and use schema based parsing. To help with debugging the failures, I also improved the error message for `RecordAttrDoesNotExist`

## Issue #, if available
Implements https://github.com/cedar-policy/cedar/issues/181 and https://github.com/cedar-policy/cedar/issues/234

## Checklist for requesting a review

The change in this PR is (choose one, and delete the other options):

- [ ] A breaking change requiring a major version bump to `cedar-policy` (e.g., changes to the signature of an existing API).
- [ ] A backwards-compatible change requiring a minor version bump to `cedar-policy` (e.g., addition of a new API).
- [x] A bug fix or other functionality change requiring a patch to `cedar-policy`.
- [ ] A change "invisible" to users (e.g., documentation, changes to "internal" crates like `cedar-policy-core`, `cedar-validator`, etc.)
- [ ] A change (breaking or otherwise) that only impacts unreleased or experimental code.

I confirm that this PR (choose one, and delete the other options):

- [x] Updates the "Unreleased" section of the CHANGELOG with a description of my change (required for major/minor version bumps).
- [ ] Does not update the CHANGELOG because my change does not significantly impact released code.

I confirm that [`cedar-spec`](https://github.com/cedar-policy/cedar-spec) (choose one, and delete the other options):

- [x] Does not require updates because my change does not impact the Cedar Dafny model or DRT infrastructure.
- [ ] Requires updates, and I have made / will make these updates myself. (Please include in your description a timeline or link to the relevant PR in `cedar-spec`, and how you have tested that your updates are correct.)
- [ ] Requires updates, but I do not plan to make them in the near future. (Make sure that your changes are hidden behind a feature flag to mark them as experimental.)
- [ ] I'm not sure how my change impacts `cedar-spec`. (Post your PR anyways, and we'll discuss in the comments.)

## Disclaimer

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by andrewmwells-amazon** [comment]
Downstream failure is expected.

**Comment by andrewmwells-amazon** [comment]
Will have to modify/re-generate the corpus for the new error message.

File: cedar-policy-core/src/evaluator/err.rs [hunk 1]
```diff
@@ -43,8 +43,8 @@ pub enum EvaluationError {
 
     /// Tried to get an attribute of a (non-entity) record, but that record
     /// didn't have that attribute
-    #[error("record does not have the attribute: {0}")]
-    RecordAttrDoesNotExist(SmolStr),
+    #[error("record does not have the attribute: {0}. Available attrs: {1:?}")]
```
Comment (khieta) [inline cedar-policy-core/src/evaluator/err.rs:46]: Consistency nit:
```suggestion
    #[error("record does not have the attribute: {0}. Available attributes: {1:?}")]
```
Also, what does the `{1:?}` syntax mean? What does this print if the list of available attributes is empty? Possibly we want a separate function (like `pretty_type_error` below) that prints something like "record has no attributes" in this case.

**Review by khieta — APPROVED** [review]

File: cedar-integration-tests/corpus_tests/9deac7d6cae0d5e05635469c985c01f9a22e0d1b.json [hunk 2]
```diff
@@ -15,7 +15,7 @@
       "decision": "Deny",
       "reasons": [],
       "errors": [
-        "error occurred while evaluating policy `policy0`: record does not have the attribute: f"
+        "error occurred while evaluating policy `policy0`: record does not have the attribute: f. Available attrs: [\"\"]"
```
Comment (khieta) [inline cedar-integration-tests/corpus_tests/9deac7d6cae0d5e05635469c985c01f9a22e0d1b.json:18]: CI is failing because this should be "attributes"

**Review by shaobo-he-aws — APPROVED** [review]

File: cedar-integration-tests/corpus_tests/9deac7d6cae0d5e05635469c985c01f9a22e0d1b.json (hunk 2 above)
Comment (andrewmwells-amazon) [reply cedar-integration-tests/corpus_tests/9deac7d6cae0d5e05635469c985c01f9a22e0d1b.json:18]: 😭

**Pull request merged**